lib/matplotlib/figure.pyi

```diff
@@ -154,15 +153,15 @@ class FigureBase(Artist):
         *,
         sharex: bool | Literal["none", "all", "row", "col"] = ...,
         sharey: bool | Literal["none", "all", "row", "col"] = ...,
-        squeeze: Literal[True] = ...,
+        squeeze: bool = ...,
```

Comment:
Why did this change?

Reply:
This is a particular case of a potentially more general issue, which was closely related enough to this change that I decided to make the change for this one here.

Essentially while it seems like `Literal[True]` and `Literal[False]` should actually span `bool`, they technically do not, as they only handle the case of literals being passed, not a variable with type `bool`.

That said, in many cases it doesn't really make _sense_ to pass anything other than a literal there, so the "passed an unknown bool" case is pretty unexpected.
e.g. the `cbook.load_data` method has a couple booleans that we would never expect to be passed as an unknown bool (in part because that is a largely for internal use method).

In this particular case a more complete answer would be to try and do our best with `overloads` to mete out the case which is passed `sqeeze=True` with `Literal[1]` for nrows/ncols and then have a fallback for this version which can't necessarily differentiate what the return will be.

In fact, unions in returns are generally discouraged and the prevailing wisdom is to just use `Any` (if overloads cannot properly distinguish the return type). This is actually what I've already done with `pyplot.subplots`, which calls this method.

Removing such unions is a change I've been leaning towards doing at some point, but made a minimal change here as the return union actually encompasses all possible returns regardless of the value of `squeeze` here.